lib/src/soloud.dart

```diff
@@ -287,6 +287,14 @@ interface class SoLoud {
       return;
     }
 
+    // if `!_isInitialized` but the engine is initialized in native, therefore
+    // the developer may have carried out a hot reload which does not imply
+    // the release of the native player.
+    // Just deinit the engine to be re-inited later.
+    if (SoLoudController().soLoudFFI.isInited()) {
+      deinit();
```

Comment:
I think this calls for a warning in logs.

```suggestion
      _log.warning("init() called when the native player is already initialized. This is expected after a hot restart but not otherwise. If you see this in production logs, there's probably a bug in your code. You may have neglected to deinit() SoLoud during the current lifetime of the app.");
      deinit();
```

I'm writing this in github web interface so the formatting is wrong but you get the idea.